bugsnag-android-core/src/main/java/com/bugsnag/android/ThreadState.kt

```diff
@@ -29,49 +28,55 @@ internal class ThreadState @Suppress("LongParameterList") @JvmOverloads construc
             (sendThreads == ThreadSendPolicy.UNHANDLED_ONLY && isUnhandled)
 
         threads = when {
-            recordThreads -> captureThreadTrace(
-                stackTraces ?: java.lang.Thread.getAllStackTraces(),
-                currentThread ?: java.lang.Thread.currentThread(),
-                exc,
-                isUnhandled,
-                projectPackages,
-                logger
-            )
+            recordThreads -> captureThreadTrace(exc, isUnhandled, maxThreads, projectPackages, logger)
             else -> mutableListOf()
         }
     }
 
     private fun captureThreadTrace(
-        stackTraces: MutableMap<java.lang.Thread, Array<StackTraceElement>>,
-        currentThread: java.lang.Thread,
         exc: Throwable?,
         isUnhandled: Boolean,
+        maxThreadCount: Int,
         projectPackages: Collection<String>,
         logger: Logger
     ): MutableList<Thread> {
-        // API 24/25 don't record the currentThread, add it in manually
-        // https://issuetracker.google.com/issues/64122757
-        if (!stackTraces.containsKey(currentThread)) {
-            stackTraces[currentThread] = currentThread.stackTrace
-        }
-        if (exc != null && isUnhandled) { // unhandled errors use the exception trace for thread traces
-            stackTraces[currentThread] = exc.stackTrace
-        }
+        val currentThread = java.lang.Thread.currentThread()
```

Comment:
We could clean these up nicely by using:
```kotlin
import java.lang.Thread as JavaThread
```

and then referring to them as `JavaThread` (or maybe `JThread`?) instead of the fully qualified thread name